client.js

```diff
@@ -1,31 +1,47 @@
-function u8tohex (arr) {
-  var vals = [ '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'a', 'b', 'c', 'd', 'e', 'f' ]
-  var ret = ''
-  for (var i = 0; i < arr.length; ++i) {
-    ret += vals[(arr[i] & 0xf0) / 0x10]
-    ret += vals[(arr[i] & 0x0f)]
+function MonetizerClient (opts) {
+  var domain = new URL(window.location).origin
+  this.url = domain
+  if (opts && opts.url) {
+    this.url = opts.url
   }
-  return ret
-}
+  this.cookieName = (opts && opts.cookieName) || '__monetizer'
+  this.receiverUrl = this.url + '/__monetizer/:id'
+
+  const COOKIE_REGEX = new RegExp(this.cookieName + '=(.*?)(;|$)')
+  this.getMonetizationId = function () {
+    const match = document.cookie.match(COOKIE_REGEX)
+    if (!match) {
+      return null
+    }
+
+    return match[1]
+  }
+
+  this.start = function () {
+    var self = this
```

Comment:
I don't think this variable is needed because you're using arrow functions for the promise